nanoc-core/lib/nanoc/core/item_rep_writer.rb

```diff
@@ -62,11 +62,25 @@ def write_single(item_rep, compiled_content_store, snapshot_name, raw_path, writ
 
         # Write
         if is_modified
+          # Do a copy-on-write (CoW) filesystem-aware reflink-copying when item_rep is identical
+          # to the source content. Fallback to a hardlink or full copy.
+          written = false
           begin
-            FileUtils.ln(temp_path, raw_path, force: true)
-          rescue Errno::EXDEV, Errno::EACCES
-            FileUtils.cp(temp_path, raw_path)
+            require 'reflink'
+            if !content.filename.nil? && FileUtils.identical?(content.filename, temp_path)
+              Reflink.reflink_always(content.filename, raw_path)
```

Comment:
I think this needs to be 

```
require 'reflink'
Reflink.reflink_always(temp_path, raw_path)
```

The `content.filename` is not necessarily set.

Reply:
(Though I am not sure what happens when `raw_path` already exists.)

Reply:
(Also, `temp_path` and `raw_path` are bad variable names. I’ll need to fix that…)

Reply:
> The `content.filename` is not necessarily set.

I check if it is set on the line above. I specifically want to reflink the source content file (content/…) and not the working copy (temp_path). I compare them for similarity on the line above. The temp_path is deleted later (and lives on another filesystem at /tmp (tempfs)) so its completely pointless to reflink or hardlink it.